libraries/ESP32/examples/Serial/Interrupt/Interrupt.ino

```diff
@@ -0,0 +1,42 @@
+#define BUFFER_SIZE 8
+
+static volatile char inputBuffer[BUFFER_SIZE];
+static volatile size_t inputBufferLength = 0;
+
+// Please keep in mind, since the ESP32 is dual core, 
+// the interrupt will be running on the same core as the setRxInterrupt function was called on.
+static void IRAM_ATTR onSerialRX(uint8_t character, void* user_arg){
+	
```

Comment:
this is a dangerous code, because the ISR might run in the middle of your for loop in `loop()`. Gotchas like this is the reason why this functionality was never added :) Using some sort of ring buffer can help, especially if it's protected by semaphore/mutex.

Reply:
Oh yeah sure, pardon me!